packages/typography/src/elements/CodeBlock.tsx

```diff
@@ -18,13 +24,41 @@ export interface ICodeBlockProps extends HTMLAttributes<HTMLPreElement> {
   isLight?: boolean;
   /** Displays line numbers */
   isNumbered?: boolean;
+  /** Determines the lines to highlight */
+  highlightLines?: number[];
+  /** Passes props to the code block container */
+  containerProps?: HTMLAttributes<HTMLDivElement>;
 }
 
 /**
  * @extends HTMLAttributes<HTMLPreElement>
  */
 export const CodeBlock = React.forwardRef<HTMLPreElement, ICodeBlockProps>(
-  ({ children, isLight, isNumbered, language, size, ...other }, ref) => {
+  (
+    { children, containerProps, highlightLines, isLight, isNumbered, language, size, ...other },
+    ref
+  ) => {
+    const [containerTabIndex, setContainerTabIndex] = useState(-1);
+    const containerRef = useRef<HTMLDivElement>(null);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const updateContainerTabIndex = useCallback(
+      debounce(() => {
+        if (containerRef.current) {
+          const codeBlock = containerRef.current.children[0];
```

Comment:
Would [firstChild](https://developer.mozilla.org/en-US/docs/Web/API/Node/firstChild) work in this case?

Reply:
Gives me a Node vs the desired Element which is dereferenced for width/height.